cmd/bucket-replication.go

```diff
@@ -91,7 +91,7 @@ func getReplicationConfig(ctx context.Context, bucketName string) (rc *replicati
 
 // validateReplicationDestination returns error if replication destination bucket missing or not configured
 // It also returns true if replication destination is same as this server.
-func validateReplicationDestination(ctx context.Context, bucket string, rCfg *replication.Config, checkRemote bool) (bool, APIError) {
+func validateReplicationDestination(ctx context.Context, bucket string, rCfg *replication.Config, opts *validateReplicationDestinationOptions) (bool, APIError) {
```

Comment:
I would suggest setting `opts` to a default structure at the beginning at this function if its value is `nil`. I'm not really sure why this boolean needed to be changed into a structure.

Reply:
Ah, then you need to keep it a pointer and not pass by value. But you could have used a `*bool` as well instead of the structure.